### PR TITLE
[RA] Visual, Polish/Performance and Balance Tweak for the next Playtest

### PR DESCRIPTION
**Pull request by SoScared**
This is the soft end on a list of changes played out on balance playtest maps: http://www.sleipnirstuff.com/forum/viewtopic.php?t=20017

With few exceptions this list focus on neutral game elements that seek to improve overall game play experience. With the Hit-Shape and Unit Stance features coming up I've held off changes that alters units and structures in relation to performance and damage vs buildings and defensive structures. Changes that crosses the effect of the HitShape/UnitStance of the upcoming official playtest ought to be held off until we're satisfied and familiar with how they work.

Most of these changes are in fact inspired from Fortnight's Overhaul mod (http://www.sleipnirstuff.com/forum/viewtopic.php?p=299841) that sought to improve the game in various ways. As a whole his mod was straying a bit too far off the deep end but with it came an endless amount of changes that brings new and interesting solutions to problems you didn't really notice.

- - - - - - - -

VISUAL:

- MiG/YAK - Simplified unit names in UI ***removed**
- Longbow - SHP fix
- Chinook - SHP fix
- Parabombs - Remove shadows

The full names MiG/Yak Attack Plane is a bit unnecessary and looks noisy in the UI in-game when attempting to hover over a moving unit. The [SHP fixes](http://www.sleipnirstuff.com/forum/viewtopic.php?p=299841#299841) has done a great job removing visible smudges on the units.

Parabomb shadows was turned on :True through a [recent commit](https://github.com/OpenRA/OpenRA/pull/13522/commits/78de7c2294c7147277b22a19fec2c5a8a43f0bcc) and I take it was to fix a glitch overlooked by #12737 however the PR left it :False deliberately because it just didn't look very good. The parabomb shadow, although physically correct, "dances" on spot until the projectile hit and takes away from the experience of watching the drop demolish its targets.

- - - - - - - -

PERFORMANCE/DESIGN:

- ^Helicopter - WaitDistanceFromResupplyBase: 4c0, up from 3c0
- Havester - Ore (From Order), 11, down from 15
- Silo - Stores 3k - Build Adjacency: 4c0, up from 2c0
- Kennel - Build adjacency: 4c0, up from 2c0
- Service Depot - 30% faster repair speed, add traversable cells
- War Factory, Soviet Tech Center, Sub Pen, Naval Yard - vision range: 5c0, up from 4c0
- Unify cloak detection range on all defensive structures - range: 6c0 (Turret -1, Tesla Coil -2)

These are all aims to smooth out the game experience in various ways. Less cloggy helicopters, harvesters that sticks more to its designated ore patch and Silos and Kennels that can be placed a bit off base so as not to obstruct general base planning and crawling. The Silo storage capacity increase actually made the structure borderline useful. Service depot with traversable cells top/bottom looked decent enough and supports better unit movement in base.

The change with vision range and cloak detection range is part of a move to streamline the game design a bit. With exception to cheap structures such as Power Plants and Silos there's no apparent reason why the War Factory and Soviet Tech Center has a vision range of 4c0 while Barracks and Service Depot has a vision range of 5c0. As for cloak detection, one important factor in using cloaked units is to be able to plan routes and estimate risks around detection. Different ranged cloak detection on defensive structures kind of discourages the attempt as players can't get a good feel of how to traverse around enemy bases safely. These are pretty minor but if nothing else it would smoothen out the game design aspect in which players can better anticipate certain behaviors from the game.

- - - - - - - - -

GAME BALANCING:

- Dog - HP: 18, up from 12
- +1c0 vision range for Medium, Heavy and Mammoth Tank
- Power-down/low-power reduces vision range down to 5c0 from 10c0 on Radar Dome, Iron Curtain and Chronosphere.
- Camo. Pillbox – cost $700, down from $800 - HP 400, down from 450
- Walls/Sandbag/Fence - No longer gives experience.
- Hospital – vision range 4, up from 3 - HP 800, down from 1000
- Oil Derrick - vision range 4, up from 3 - HP 800, down from 1000
- Civilian/Technician – vision range 3, up from 2.

The dogs HP change doesn't take into account #13513 but as that PR aims for the next release I'd love to include this change in the playtest. Despite being bugged the tiny HP increase has extended the use of the dog, especially by players using it as scouts. Basically it can absorb one hit from the rifleman making it win 1v1 engagements and provides a window of escape when passing the edge of the rifleman's vision range.

+1 vision range for the main battle tanks promotes better army retention, meaning players are better equipped to decide wether or not to engage with their units supported by tanks. In practice this allowed players to use their armies more towards grandstanding and map control as opposed to go in blind when the enemy is detected. In turn players felt a bit more comfortable teching up behind their Service Depot army.

The Camo.Pillbox Price/HP change - having equal HP as the regular Pillbox makes memorization and intuitive action by players better. The downscaling of the price reduction vs HP reduction is fairly similar.

The 1000 HP on the Oil Derrics and Hospitals has always felt just a little too much tanky and time consuming to kill off. 800 HP felt more sensible and the +1c0 vision enabled the structures to function better as strategic points in addition to their perks.

The Technician vision range 2c0 simply looks bad in-game and with 3c0 can actually be used as map spotters.

- - - - - - -

Overall these won't really change game balancing all too much but will most definitely level up the game experience for the playerbase. As initially stated game balancing beyond the official playtest ought to be brought onboard after its release as the HitShape/UnitStance risk making previous game balance alterations moot. This might feel like a step back in the ongoing game balancing process for RA but it seems to me if the official playtest turns out to be successful the rewards can outweigh the hurdles many times over.

### CODE REVIEW

File: mods/ra/weapons/superweapons.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ ParaBomb:
 		OpenSequence: open
 		Velocity: 0, 0, -86
 		Acceleration: 0, 0, 0
-		Shadow: True
+		Shadow: False
```
Comment (abcdefg30) [inline mods/ra/weapons/superweapons.yaml]: This as well.

Comment (abcdefg30) [inline mods/ra/weapons/superweapons.yaml]: Ah nvm, I see you changed that back intentionally.

File: mods/ra/rules/structures.yaml [hunk 2]
```diff
@@ -1719,7 +1725,7 @@ FIX:
 	Tooltip:
 		Name: Service Depot
 	Building:
-		Footprint: _X_ xxx _X_
+		Footprint: _=_ xxx _=_
```
Comment (SoScared) [inline mods/ra/rules/structures.yaml]: The top side of the SD looks to be perfectly left alone. The bottom is slightly run over but doesn't look intrusive as units are crossing a flat structure. IMO a good trade-off for mobility.

**Comment by FrameLimiter** [comment]
∙The HBOX has a CustomSellValue of 600. It should be changed to 500 because of its new price.
∙The SAM has a DetectCloaked range of 5 cells. It should be changed to 6 to match the other defenses.

**Comment by SoScared** [comment]
Fixed! I had cloak detection for the SAM and AGUN removed on the playtest maps but the pillbox customsellvalue always gets me.

File: mods/ra/rules/structures.yaml [hunk 3]
```diff
@@ -570,9 +570,13 @@ DOME:
 		Type: Wood
 	RevealsShroud:
 		Range: 10c0
+		RequiresCondition: !disabled
 		RevealGeneratedShroud: False
+	RevealsShroud@Offline:
+		Range: 5c0
```
Comment (reaperrr) [inline mods/ra/rules/structures.yaml]: Nit: I know this won't have any gameplay impact, as it's hidden by the other ones as long as they are enabled, but for performance reasons I'd give this `RequiresCondition: disabled`, to disable it while the other two are enabled (our engine probably does visibility updates for every enabled trait, regardless of whether there's one with higher Range that 'drowns' it).

Comment (SoScared) [inline mods/ra/rules/structures.yaml]: Well the PR title mentions 'performance' after all :)

Added!

**Comment by AoAGeneral** [comment]
Should concrete walls be a requirement for war factory? Currently you can skip sandbags/wire fences since the game starts you with concrete walls from the get go.

**Comment by reaperrr** [comment]
> Should concrete walls be a requirement for war factory?

I assume you meant it the other way round?

**Review by pchote — COMMENTED** [review]
I'm not a fan of the Mig/Yak rename, but the other changes seem reasonable.
The tooltip width is fixed by the player name/faction, and so reducing the names to 3 characters then just leaves a large amount of unpolished-looking empty space in the tooltip.

**Comment by SoScared** [comment]
I could drop the renaming in a heartbeat but are you sure? On a related note, would there be possible to have different tooltips for the build bar and units on field?

**Comment by SoScared** [comment]
Done!

**Review by pchote — APPROVED** [review]

**Comment by reaperrr** [comment]
👍

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/120978f37f856f4ac9b52857b00e7aecbb9f4c7c...63a60e07d6641aaef4d722282988a3cb56e9231b

**Comment by SoScared** [comment]
I missed the power-down vision reduction for the Iron Curtain and Chronosphere in the code. Putting up a new PR asap.

**Pull request merged**